Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -76,9 +76,11 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
 
     var context: RUMContext {
         var context = parent.context
-        context.activeViewID = viewUUID
-        context.activeViewURI = viewURI
-        context.activeUserActionID = userActionScope?.actionUUID
+        if isActiveView {
+            context.activeViewID = viewUUID
+            context.activeViewURI = viewURI
+            context.activeUserActionID = userActionScope?.actionUUID
+        }
```

Comment:
I'm not so sure on this 💭. The `context` propagates to child scopes which in case of `RUMViewScope` are: `RUMResourceScope` and `RUMUserActionScope`.

The `RUMResourceScope` captures the view's `context` by value:
```swift
// RUMResourceScope.swift

let context: RUMContext
```
so it's immutable and not impacted by this change.

However, the `RUMUserActionScope` references view's `context` using `parent`:
```swift
// RUMUserActionScope.swift

private unowned let parent: RUMContextProvider // it's set to `RUMViewScope`

var context: RUMContext { parent.context }
```
so it will receive session scope's `context` if the view scope becomes inactive.

Oh the other hand, following the implementation of `RUMViewScope` and `RUMUserActionScope` the view's `context` seems to be not read by action scope after the view becomes inactive. What was the reason for this change?

Reply:
Thanks for the context (no pun). The reason was mainly for consistency in case `RUMViewScope.context` was called somehow after the view became inactive, but I wasn't sure how/if it would happen.